libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -218,6 +218,23 @@ double WaveTrack::Interval::GetPlayStartTime() const
    return mpClip->GetPlayStartTime();
 }
 
+double WaveTrack::Interval::GetPlayEndTime() const
+{
+   // TODO wide wave tracks:  assuming that all 'narrow' clips share common
+   // beginning
+   return mpClip->GetPlayEndTime();
+}
```

Comment:
You don't need this function.  Instead use `End` inherited from `ChannelGroupInterval`.

Reply:
`ChannelGroupInterval::mStart` and `ChannelGroupInterval::mEnd` are immutable, whereas `WaveClip::GetPlayStartTime` and `WaveClip::GetPlayEndTime` always return actual clip borders, though the difference may not be relevant in this particular PR

Reply:
I've addressed it in a way @Paul-Licameli suggested, but that can be undone easily.

Reply:
Yes `mStart` and `mEnd` are immutable, but now I dislike that choice I made and may change it so that a virtual function reports start and end.

That is what bothers me when I consider #5159 which I have not approved yet but isn't actually critical for release now.

For the short-lived usage of intervals in Disjoin, it doesn't matter.

Reply:
> That is what bothers me when I consider https://github.com/audacity/audacity/issues/5159

I think Vitaly has addressed the issue you asked. It would be great if that PR is reviewed, not thrown away.